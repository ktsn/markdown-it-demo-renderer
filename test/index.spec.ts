import MarkdownIt = require('markdown-it')
import deindent = require('deindent')
import pug = require('pug')
import plugin = require('../src/index')

describe('Demo Renderer', () => {
  it('should render demo container over fence', () => {
    const md = new MarkdownIt().use(plugin)

    const code = deindent(`
      \`\`\`html
      <p>Hello!</p>
      \`\`\`
    `)

    expect(md.render(code)).toMatchSnapshot()
  })

  it('should rewrite container class name', () => {
    const md = new MarkdownIt().use(plugin, {
      demoContainerClass: 'test-class-name'
    })

    const code = deindent(`
      \`\`\`html
      <h1>Test</h1>
      \`\`\`
    `)

    expect(md.render(code)).toMatchSnapshot()
  })

  it('should not render when unsupported language', () => {
    const md = new MarkdownIt().use(plugin)

    const code = deindent(`
      \`\`\`js
      function sum(a, b) {
        return a + b
      }
      \`\`\`
    `)

    expect(md.render(code)).toMatchSnapshot()
  })

  it('should be specified preprocessor', () => {
    const md = new MarkdownIt().use(plugin, {
      languages: {
        pug: pug.render
      }
    })

    const code = deindent(`
      \`\`\` pug
      h1.title Hello World!
      \`\`\`
    `)

    expect(md.render(code)).toMatchSnapshot()
  })
})
