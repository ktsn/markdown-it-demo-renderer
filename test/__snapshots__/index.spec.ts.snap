// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo Renderer should be specified preprocessor 1`] = `
"<div class=\\"markdown-demo\\"><h1 class=\\"title\\">Hello World!</h1></div><pre><code class=\\"language-pug\\">h1.title Hello World!
</code></pre>
"
`;

exports[`Demo Renderer should not render when unsupported language 1`] = `
"<pre><code class=\\"language-js\\">function sum(a, b) {
  return a + b
}
</code></pre>
"
`;

exports[`Demo Renderer should render demo container over fence 1`] = `
"<div class=\\"markdown-demo\\"><p>Hello!</p>
</div><pre><code class=\\"language-html\\">&lt;p&gt;Hello!&lt;/p&gt;
</code></pre>
"
`;

exports[`Demo Renderer should rewrite container class name 1`] = `
"<div class=\\"test-class-name\\"><h1>Test</h1>
</div><pre><code class=\\"language-html\\">&lt;h1&gt;Test&lt;/h1&gt;
</code></pre>
"
`;
